pkg/dependencies/factory.go

```diff
@@ -104,8 +104,10 @@ func ForSpec(ctx context.Context, clusterSpec *cluster.Spec) *Factory {
 }
 
 type Factory struct {
-	executableBuilder        *executables.ExecutableBuilder
+	executableBuilder        *executables.ExecutablesBuilder
 	executablesImage         string
+	executablesInDocker      bool
```

Comment:
If we have a `WithDocker`, why not call this something like `containerExecutables`, and then if we have `WithPodman` in the future, it would use the same logic here? Just want to see if that is possible vs calling everything `executableInDocker` or `InitInDocker` or something, but I understand that it would be more refactoring and something we can pick up later when we actually support the different ones.

Reply:
`containerExecutables` seems a bit ambiguous. If you prefer `executablesInContainer`, I'm ok with that.

However, as you are saying, if we want to support more container runtimes/clients, there would more stuff we would need to change.